lib/client.js

```diff
@@ -190,9 +190,9 @@ function MqttClient (streamBuilder, options) {
   })
 
   // resubscribe
-  this.on('connect', function () {
-    if (this.options.clean && Object.keys(this._subscribedTopics).length > 0) {
-      this.subscribe(this._subscribedTopics)
+  this.on('reconnect', function () {
```

Comment:
otherwise, the `subscribe` packet could be sent twice on the initial connect: once in the initial subscribe (prior to `connack`) and then subsequently during the connect event (since it was added to the `_resubscribeTopics` at initial subscribe time.